elyra/pipeline/tests/test_processor_kfp.py

```diff
@@ -230,42 +235,63 @@ def test_processing_url_runtime_specific_component(monkeypatch, processor, sampl
 
     # Construct single-operation pipeline
     pipeline = Pipeline(id='pipeline-id',
-                        name='untitled',
+                        name='kfp_test',
                         runtime='kfp',
                         runtime_config='test',
                         source='filter_text.pipeline')
     pipeline.operations[operation.id] = operation
 
-    # Process pipeline with a scaled-down version of the processor cc_pipeline function
-    constructed_pipeline = processor._cc_pipeline(pipeline=pipeline, pipeline_name='test_pipeline')
+    # Establish path and function to construct pipeline
+    pipeline_path = os.path.join(tmpdir, 'kfp_test.yaml')
+    constructed_pipeline_function = lambda: processor._cc_pipeline(pipeline=pipeline, pipeline_name='test_pipeline')
+
+    # TODO Check against both argo and tekton compilations
+    # Compile pipeline and save into pipeline_path
+    kfp_argo_compiler.Compiler().compile(constructed_pipeline_function, pipeline_path)
 
-    # Ensure the pipeline operation was properly processed for execution, creating a KFP TaskSpec object
-    assert isinstance(constructed_pipeline[operation.id], TaskSpec)
-    assert constructed_pipeline[operation.id].component_ref.url == url
+    # Read contents of pipeline YAML
+    with open(pipeline_path) as f:
+        pipeline_yaml = yaml.safe_load(f.read())
+
+    # Check the pipeline file contents for correctness
+    pipeline_template = pipeline_yaml['spec']['templates'][0]
+    assert pipeline_template['metadata']['annotations']['pipelines.kubeflow.org/task_display_name'] == operation_name
+    assert pipeline_template['inputs']['artifacts'][0]['raw']['data'] == operation_params['text']
+
+    component_ref = pipeline_template['metadata']['annotations']['pipelines.kubeflow.org/component_ref']
+    component_ref = ast.literal_eval(component_ref)
+    assert component_ref['url'] == url
 
 
-@pytest.mark.skip(reason="failing - need to evaluate function at kfp compiler level")
 def test_processing_filename_runtime_specific_component(monkeypatch, processor, sample_metadata, tmpdir):
+    # Assign test resource location
+    relative_path = "kfp/filter_text_using_shell_and_grep.yaml"
+
     # Instantiate a file-based component
     component = Component(id="filter-text",
                           name="Filter text",
                           description="",
                           op="filter-text",
                           source_type="filename",
-                          source="kfp/filter_text_using_shell_and_grep/component.yaml",
+                          source=relative_path,
                           properties=[],
                           catalog_entry_id="")
 
     # Replace cached component registry with single filename-based component for testing
     processor._component_registry._cached_components = [component]
 
     # Construct hypothetical operation for component
+    operation_name = "Filter text test"
```

Comment:
Given this component is remaining in its present location and (absolutely) doesn't contain a " test" suffix, I suspect this answers my previous questions and `operation_name` is just a rendered entity.